4.sc_Morphology_Neural_Network_MLP_Model/0.model_training/scripts/Hyperparameter_Optimization_model_binary.py

```diff
@@ -428,56 +454,63 @@
 )
 df_values_Y = df_values["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"]
 
-# %%
-test_data = Dataset_formatter(
-    torch.FloatTensor(df_values_X.values), torch.FloatTensor(df_values_Y.values)
-)
-
-# convert data class into a dataloader to be compatible with pytorch
-test_loader = torch.utils.data.DataLoader(dataset=test_data, batch_size=1)
+# %% [markdown]
+# ## Test the hold out wells
 
 # %%
-# calling the testing function and outputting list values of tested model
-if params.MODEL_TYPE == "Multi_Class" or params.MODEL_TYPE == "Regression":
-    y_pred_list = test_optimized_model(
-        model, test_loader, params, model_name=params.MODEL_NAME
+for SHUFFLE in [True, False]:
+    test_data = Dataset_formatter(
+        torch.FloatTensor(df_values_X.values), torch.FloatTensor(df_values_Y.values)
     )
-elif params.MODEL_TYPE == "Binary_Classification":
-    y_pred_list, y_pred_prob_list = test_optimized_model(
-        model, test_loader, params, model_name=params.MODEL_NAME
-    )
-else:
-    raise Exception("Model type must be specified for proper model testing")
 
+    # convert data class into a dataloader to be compatible with pytorch
+    test_loader = torch.utils.data.DataLoader(
+        dataset=test_data, batch_size=1, shuffle=SHUFFLE
+    )
 
-# un-nest list if nested i.e. length of input data does not match length of output data
-if len(y_pred_list) != len(df_values_Y):
-    y_pred_list = un_nest(y_pred_list)
-    y_pred_prob_list = un_nest(y_pred_prob_list)
-else:
-    pass
+    # calling the testing function and outputting list values of tested model
+    if params.MODEL_TYPE == "Multi_Class" or params.MODEL_TYPE == "Regression":
+        y_pred_list = test_optimized_model(
+            model, test_loader, params, model_name=params.MODEL_NAME
+        )
+    elif params.MODEL_TYPE == "Binary_Classification":
+        y_pred_list, y_pred_prob_list = test_optimized_model(
+            model, test_loader, params, model_name=params.MODEL_NAME
+        )
+    else:
+        raise Exception("Model type must be specified for proper model testing")
+
+    # un-nest list if nested i.e. length of input data does not match length of output data
+    if len(y_pred_list) != len(df_values_Y):
+        y_pred_list = un_nest(y_pred_list)
+        y_pred_prob_list = un_nest(y_pred_prob_list)
+    else:
+        pass
+
+    # Call visualization function
+    # calling the testing function and outputing list values of tested model
+    if params.MODEL_TYPE == "Multi_Class" or params.MODEL_TYPE == "Regression":
```

Comment:
```suggestion
    if any(model_type == params.MODEL_TYPE for model_type in ["Multi_Class", "Regression"]):
```
use the `any()` to avoid duplication